src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,205 @@
+  private Integer playerBracketIndex = null;
+  @Setter
+  private double maxWidth = 0;
+  @Setter
+  private double maxHeight = 0;
+
+
+  @Override
+  protected void onInitialize() {
+    JavaFxUtil.bindManagedToVisible(loadingPane);
+  }
+
+  @Override
+  public Pane getRoot() {
+    return root;
+  }
+
+
+  public void setQueue(MatchmakerQueueInfo queue) {
+    mapService.getMatchmakerBrackets(queue).subscribe(rawBrackets -> {
+      loadingPane.setVisible(false);
+
+      this.sortedBracketsWithDuplicates = this.getSortedBrackets(rawBrackets);
+
+      this.sortedBrackets = new TreeMap<>(this.sortedBracketsWithDuplicates);
+      this.sortedBrackets.replaceAll((key, value) -> new ArrayList<>(value));
+      this.removeDuplicates(this.sortedBrackets);
+
+      PlayerInfo player = playerService.getCurrentPlayer();
+      LeaderboardRating ratingBean = player.getLeaderboardRatings().get(queue.getLeaderboard().technicalName());
+      double rating = (ratingBean != null) ? (ratingBean.mean() - 3 * ratingBean.deviation()) : 0;
+      this.playerBracketIndex = this.getPlayerBracketIndex(this.sortedBrackets, rating);
+
+      List<MapVersion> values = this.sortedBrackets.values().stream().flatMap(List::stream).toList();
+      this.resizeToContent(values.size(), TILE_SIZE);
+
+
+      fxApplicationThreadExecutor.execute(() -> values.forEach(this::addMapTile));
+    });
+
+  }
+
+
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> getSortedBrackets(Map<MatchmakerQueueMapPool, List<MapVersion>> brackets) {
+
+    Comparator<MapVersion> mapVersionComparator = Comparator.nullsFirst(Comparator.comparing(MapVersion::size))
+                                                            .thenComparing(mapVersion -> mapVersion.map().displayName(),
+                                                                           String.CASE_INSENSITIVE_ORDER);
+
+    Comparator<MatchmakerQueueMapPool> mapPoolComparator = Comparator.comparing(MatchmakerQueueMapPool::minRating, Comparator.nullsFirst(Double::compare))
+                                                                     .thenComparing(MatchmakerQueueMapPool::maxRating, Comparator.nullsLast(Double::compare));
+
+    SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedMap = new TreeMap<>(mapPoolComparator);
+
+    brackets.forEach((mapPool, mapVersions) -> {
+      List<MapVersion> sortedList = mapVersions.stream().sorted(mapVersionComparator).toList();
+      sortedMap.put(mapPool, sortedList);
+    });
+
+    return sortedMap;
+  }
+
+  private void removeDuplicates(SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets) {
+    Set<String> usedMaps = new HashSet<>();
+    sortedBrackets.replaceAll((k, v) -> sortedBrackets.get(k).stream().filter(mapVersion -> {
+      String name = mapVersion.map().displayName();
+      if (usedMaps.contains(name)) {
+        return false;
+      }
+      usedMaps.add(name);
+      return true;
```

Comment:
The sortedBrackets.get(k) is unnecessary you already have access to the value. Also please name the variables appropriately.

Reply:
Also rather than replacing it would be better to just return a new map. Then you don't have to copy all the lists as well when you create sorted brackets.